File: frontend/src/types/attempt.ts
export interface PerQuestionResult {
  question_id: string
  question_number: number
  question_text: string
  user_answer: string | null
  correct_answer: string
  is_correct: boolean
  explanation: string
}

export interface AttemptResponse {
  id: string
  quiz_id: string
  score: number
  total_questions: number
  correct: number
  incorrect: number
  per_question: PerQuestionResult[]
  completed_at: string
}

export interface AttemptSummary {
  id: string
  quiz_id: string
  score: number | null
  completed_at: string
}

export interface AttemptSubmitRequest {
  quiz_id: string
  answers: Record<string, string>
}

File: backend/app/schemas/attempt.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class AttemptSubmitRequest(BaseModel):
    quiz_id: uuid.UUID
    answers: dict[str, str]  # {question_id: answer}


class PerQuestionResult(BaseModel):
    question_id: uuid.UUID
    question_number: int
    question_text: str
    user_answer: str | None
    correct_answer: str
    is_correct: bool
    explanation: str


class AttemptResponse(BaseModel):
    id: uuid.UUID
    quiz_id: uuid.UUID
    score: float
    total_questions: int
    correct: int
    incorrect: int
    per_question: list[PerQuestionResult]
    completed_at: datetime

    model_config = {"from_attributes": True}


class AttemptSummaryResponse(BaseModel):
    id: uuid.UUID
    quiz_id: uuid.UUID
    score: float | None
    completed_at: datetime

    model_config = {"from_attributes": True}


class AttemptListResponse(BaseModel):
    items: list[AttemptSummaryResponse]
    total: int
    page: int
    size: int
